### PR TITLE
Bug/merge arrays

### PR DESCRIPTION
**Pull request by dcalvoalonso**
Fixes https://github.com/telefonicaid/iotagent-node-lib/issues/569

### CODE REVIEW

File: lib/services/devices/deviceService.js [hunk 1]
```diff
@@ -260,11 +260,21 @@ function mergeDeviceWithConfiguration(fields, defaults, deviceData, configuratio
```
Comment (fgalan) [inline lib/services/devices/deviceService.js]: CHANGE_NEXT_RELEASE entry (mentioning #569) missing

File: test/unit/lazyAndCommands/lazy-devices-test.js [hunk 2]
```diff
@@ -102,6 +122,21 @@ var iotAgentLib = require('../../../lib/fiware-iotagent-lib'),
         type: 'Motion',
         service: 'smartGondor',
         subservice: 'gardens'
+    },
+    device3 = {
+        id: 'TestRobotPre',
+        type: 'RobotPre',
+        service: 'smartGondor',
```
Comment (fgalan) [inline test/unit/lazyAndCommands/lazy-devices-test.js]: Should be "smartgondor" ?

File: lib/services/devices/deviceService.js (hunk 1 above)
Comment (dcalvoalonso) [reply lib/services/devices/deviceService.js]: Fixed in 63fe182ea98e6280914a7c6a4366550762f8a636

File: test/unit/lazyAndCommands/lazy-devices-test.js (hunk 2 above)
Comment (dcalvoalonso) [reply test/unit/lazyAndCommands/lazy-devices-test.js]: Yes, we should follow the recommendation of using service names in lowercase. Nevertheless, in order to be consistent with the rest of the tests, I have maintained smartGondor.

You can check this in https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/lazyAndCommands/lazy-devices-test.js#L97, and https://github.com/telefonicaid/iotagent-node-lib/blob/master/test/unit/lazyAndCommands/lazy-devices-test.js#103. It is also the same in the rest of tests.

Comment (fgalan) [reply test/unit/lazyAndCommands/lazy-devices-test.js]: Ok, let's use smartGondor to keep style. NTC

File: CHANGES_NEXT_RELEASE [hunk 3]
```diff
@@ -1 +1,2 @@
 - Upgrade mongoose dependence version to 4.13.3
+- Fix bug in mergeArrays function of deviceService (#571)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: Looking to the changes in the code it seems the fix is about internalAttributes, so, maybe the following would be more precise?

> Fix bug in mergeArrays function of deviceService, precluding updates to work when internalAttributes of type Array were used (#571)

Comment (dcalvoalonso) [reply CHANGES_NEXT_RELEASE]: Fixed in 9fa97a4d08a4b2abc620958d5455a2afcd313531

File: test/unit/plugins/multientity-plugin_test.js [hunk 4]
```diff
@@ -104,7 +104,7 @@ var iotAgentLib = require('../../../lib/fiware-iotagent-lib'),
```
Comment (fgalan) [inline test/unit/plugins/multientity-plugin_test.js]: This file seems not been in the original PR... maybe it has been leaked in the last commit?

Comment (dcalvoalonso) [reply test/unit/plugins/multientity-plugin_test.js]: Sorry!!! You are right! Fixed in c7ac799e0cc31ee2326f2f81bf29fc541f94eb26

**Comment by AlvaroVega** [comment]
LGTM

**Comment by fgalan** [comment]
LGTM

**Pull request merged**